terraform-modules/cloudsql-mysql/outputs.tf

```diff
@@ -2,3 +2,7 @@
 output "cloudsql-public-ip" {
   value = "${google_sql_database_instance.cloudsql-instance.0.first_ip_address}"
 }
+
+output "cloudsql-instance-name" {
+  value = "${google_sql_database_instance.cloudsql-instance.0.name}"
```

Comment:
This reminded me about an issue I ran into that I neglected to do a PR for.  Essentially on outputs like above - where we explicitly reference element 0 of a resource.  If count is zero - ie database does not exist but module is used you will get an TF error. (something like resource not found).  Here is one reference to this: https://github.com/hashicorp/terraform/issues/16681 (there was another I came across but  can not find URL)

The URL suggests the following: 

value = "${element(concat(google_sql_database_instance.cloudsql-instance.*.name, list("")), 0)}"

I used the following that came from another another web reference:

value = "${ join("", google_sql_database_instance.cloudsql-instance.*.first_ip_address) }"

The join seems a little easier to read to me.   Here is a TF 0.11 upgrade guide that kind of explains things:
https://www.terraform.io/upgrade-guides/0-11.html#referencing-attributes-from-resources-with-count-0

If I recall I ran into this when I created a db and then I set count to 0 and destroyed the instance.  Subsequent plans would complain about missing resource.

Reply:
It sounded like you decided against these changes when we talked last week. If you'd still like me to do that  let me know and I'll put in a new PR; for now I'm going to merge this one.